docs/deployment-raft.md

```diff
@@ -0,0 +1,115 @@
+  - You may only interact with the _leader_. A good way to achieve this is using a proxy.
+- Use the [orchestrator-client](orchestrator-client.md) script.
+  - Deploy `orchestrator-client` on any box from which you wish to interact with `orchestrator`.
+  - Create and edit `/etc/profile.d/orchestrator-client.sh` on those boxes to read:
+    ```
+    ORCHESTRATOR_API="http://your.orchestrator.service.proxy:80/api"
+    ```
+    or
+    ```
+    ORCHESTRATOR_API="http://your.orchestrator.service.host1:3000/api http://your.orchestrator.service.host2:3000/api http://your.orchestrator.service.host3:3000/api"
+    ```
+    In the latter case you will provide the list of all `orchestrator` nodes, and the `orchetsrator-client` script will automatically figure out which is the leader. With this setup your automation will not need a proxy (though you may still wish to use a proxy for web interface users).
+
+    Make sure to chef/puppet/whatever the `ORCHESTRATOR_API` value such that it adapts to changes in your environment.
+
+You may _not_ use the `orchestrator` command line client. Fortunately `orchestrator-client` provides an almost identical interface as the command lien client.
+
+### Orchestrator service
+
+As noted, a single `orchestrator` node will assume leadership. Only the leader will:
+
+- Run recoveries
+
+However all nodes will:
+
+- Discover (probe) your MySQL topologies
+- Run failure detection
+- Register their own health check
+
+None-leader nodes may _NOT_:
+
+- Run arbitrary commands (e.g. `relocate`, `begin-downtime`)
+- Run recoveries per human request.
+- Serve client HTTP requests (but some endpoints, such as load-balancer and health checks, are valid).
+
+### A visual example
+
+![orchestrator deployment, raft](images/orchestrator-deployment-raft.png)
+
+In the above there are three `orchestrator` nodes in a `raft` cluster, each using its own dedicated database (either `MySQL` or `SQLite`).
+
+`orchestrator` nodes communicate with each other.
+
+Only one `orchestrator` node is the leader.
+
+All `orchestrator` nodes probe the entire `MySQL` fleet. Each `MySQL` server is probed by each of the `raft` members.
+
+### orchestrator/raft operational scenarios
+
+##### A node crashes:
+
+Start the node, start the `MySQL` service if applicable, start the `orchestrator` service. The `orchestrator` service should join the `raft` group, catch up with `raft` replication log and continue as normal.
+
+##### A node crashes for too long, or a node is re-provisioned:
+
+The `raft` replication log is only kept for a couple days. If a node goes offline for longer than that, or if it completely loses data, you will need to recreate the data:
+
+- Stop the `orchestrator` service (or it should automatically PANIC after 1 minute).
+- Copy backend DB data:
+  - If `MySQL`, run backup/restore, either logical or physical.
+  - If `SQLite`, run `.dump` + restore, see [10. Converting An Entire Database To An ASCII Text File](https://sqlite.org/cli.html).
+
```

Comment:
* It is not possible to "bootstrap" a new empty orchestrator node into a raft cluster just by talking to the cluster? while using the equivalent of dumping the sqlite or mysql db backend to the new node will work it _feels_ attractive to be able to pull this info down from the cluster directly.
* I may have missed this: but are access "credentials" required to join the cluster?